url49526af437_Maria_G_CS325_tgz-pJ8-assignments_farmer_FarmerImplJ8/src/frs/hotgammon/view/Convert.java

```diff
@@ -0,0 +1,145 @@
+package frs.hotgammon.view;
+
+import frs.hotgammon.framework.*;
+
+import java.awt.Rectangle;
+import java.awt.Point;
+import java.util.*;
+
+/** Mappings between graphical coordinate (x,y) and locations;
+ * and vice versa.
+ * 
+   This source code is from the book 
+     "Flexible, Reliable Software:
+       Using Patterns and Agile Development"
+     published 2010 by CRC Press.
+   Author: 
+     Henrik B Christensen 
+     Computer Science Department
+     Aarhus University
+   
+   This source code is provided WITHOUT ANY WARRANTY either 
+   expressed or implied. You may study, use, modify, and 
+   distribute it for non-commercial purposes. For any 
+   commercial use, see http://www.baerbak.com/
+ */
+public class Convert {
+  /** Given coordinate (x,y) on the graphical backgammon board
+   * return the location it is witin, or null if not on any
+   * location.
+   * @param x the x coordinate
+   * @param y the y coordinate
+   */
+  public static Location xy2Location(int x, int y) {
+    Set<Location> s = mapLocation2Rectangle.keySet();
+    for( Location l : s ) {
+      Rectangle r = mapLocation2Rectangle.get(l);
+      if (r.contains(x,y)) {
+        return l;
+      }
+    }
+    return null;
+  }
+
+  /** Given a  location and a count of checkers on this location,
+   * compute the (x,y) position of a properly positioned checker
+   * (i.e. properly aligned).
+   * @param location the location on the board
+   * @param count the number of checkers already on this location
+   */
+  public static Point locationAndCount2xy(Location location, int count) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.